http-client/Network/HTTP/Client/Manager.hs

```diff
@@ -216,7 +217,7 @@ connKey req =
             | secure req -> simple CKSecure
             | otherwise -> simple CKRaw
         Just p
-            | secure req -> CKProxy
+            | secure req && proxySecureMode req == ProxySecureConnect -> CKProxy
```

Comment:
Similar request here about pattern matching to catch increases in data constructors. Some comment, or well named helper function, may also be helpful to clarify why we make a special secure connection only when both `secure` is true _and_ we're using the `Connect` mode.